csharp/src/Ice/Internal/RouterInfo.cs

```diff
@@ -266,22 +266,14 @@ internal void destroy()
         }
     }
 
-    //
-    // Returns router info for a given router. Automatically creates
-    // the router info if it doesn't exist yet.
-    //
-    public RouterInfo get(Ice.RouterPrx rtr)
+    // Returns router info for a given router. Automatically creates the router info if it doesn't exist yet.
+    internal RouterInfo get(Ice.RouterPrx router)
     {
-        if (rtr == null)
+        if (router is null)
         {
             return null;
         }
 
-        //
-        // The router cannot be routed.
```

Comment:
This code did not make sense. If you pass around a routed router, we should not fix silently in some internal method.

Reply:
Should we at least check to make sure that it doesn't have a sub-router?

Reply:
Restore clearing logic. It was not practical to do it elsewhere.